oak_utils/src/lib.rs

```diff
@@ -180,27 +198,96 @@ pub fn compile_protos_without_services<P>(inputs: &[P], includes: &[P])
 where
     P: AsRef<std::path::Path>,
 {
-    compile_protos_with(false, inputs, includes);
+    compile_protos_with(&mut prost_build::Config::new(), inputs, includes);
 }
 
-fn compile_protos_with<P>(generate_services: bool, inputs: &[P], includes: &[P])
+fn compile_protos_with<P>(prost_config: &mut prost_build::Config, inputs: &[P], includes: &[P])
 where
     P: AsRef<std::path::Path>,
 {
+    compile_protos_with_internal(
+        prost_config
+            .type_attribute(".", "#[derive(::oak::handle::HandleVisit)]")
+            .extern_path(".oak.handle", "::oak::handle")
+            .extern_path(".oak.encap.GrpcRequest", "::oak::grpc::GrpcRequest")
+            .extern_path(".oak.encap.GrpcResponse", "::oak::grpc::GrpcResponse"),
+        inputs,
+        includes,
+    )
+}
+
+fn compile_protos_with_internal<P>(
```

Comment:
The various types of forwarding between variant functions has become quite tangled (to the extent that I had to draw a map to figure out what was going where).  

Could we maybe restructure to use some kind of options/config structure?  Then we could keep the various `pub fn`s as-is, but have all of them forward to the same inner function that uses the option structure.
```Rust
struct CodegenOptions {
    /// If set, indicates output directory. Crate's `OUT_DIR` is used if `None`
    out_dir: Option<std::path::PathBuf>,
    /// Whether to generate Oak-specific Handle mapping code.
    generate_oak_handle_mapping: bool,
    /// Whether to generate client and server stub code for gRPC service definitions, suitable for
    /// use inside an Oak Node (i.e.  *not* code that is suitable for use in a normal application
    /// running on the host platform).
    generate_oak_grpc_services: bool,
    /// Whether to generate "normal" client stub code for gRPC service definitions, i.e. code that
    /// is suitable for use on the host platform outside of Oak.
    generate_external_grpc_service_client: bool,
    /// Whether to generate "normal" server stub code for gRPC service definitions, i.e. code that
    /// is suitable for use on the host platform outside of Oak.
    generate_external_grpc_service_server: bool,
}
```
(cf. #1093)

(Fine for this to be deferred to a later PR.)

Reply:
I fully agree this is getting out of hand. I'd prefer to keep it out of this PR since it is already a large changeset, but would be happy to add this afterwards 😄